saleor/graphql/shop/schema.py

```diff
@@ -40,9 +50,22 @@ class ShopQueries(graphene.ObjectType):
     def resolve_shop(self, _info):
         return Shop()
 
-    @load_site_callback
-    def resolve_order_settings(self, _info, site):
-        return site.settings
+    def resolve_order_settings(self, _info):
+        channel = (
+            channel_models.Channel.objects.filter(is_active=True)
```

Comment:
Shouldn't it use dataloader?

Reply:
I'm not sure about using dataloader here. As I need to find proper slug, and maybe no need to load all channels.
Or do you mean to hit database twice, first to find `slug`, second time via dataloader load the channel data?

Reply:
Right, we are fetching the first channel ordered by `slug`. I don't think that we are able to use any existing channel dataloader here.
@fowczarek WDYT?

Reply:
IMO we could add `ChannelBySlug`  dataloader. In `then` we could checkout if it's active or not.

Reply:
I'm not sure if I understand the proposed solution. Cause we still need to know the `slug` to call `ChannelBySlug` dataloader. So you suggest to hit database for that slug and then reuse this dataloader?

Reply:
No, you're right. This is the deprecated field, we could use db query here.
I didn't see any reason for multiple times fetching this field in one query.

Reply:
I am still not sure about sorting by `slug` not `pk`. We should always display results for the same objects. When someone creates a channel with slug `aaa` then we will return these fields for this channel, not the previous ones.

Reply:
@fowczarek @kadewu  Yeah, and this is an expected way, as we want to give you a possibility to create such channel, only to have an impact on these settings. Also I don't think that we should worry about this as described case can happen only when you're mix-in old & new fields.

Reply:
So we're marking the conversation as resolved, @fowczarek @kadewu  agree?